app/code/community/Bolt/Boltpay/Helper/Api.php

```diff
@@ -316,6 +320,54 @@ protected function getRatesDebuggingData($rates) {
         return $rateDebuggingData;
     }
 
+    protected function isDiscountRoundingDeltaError($transaction, $quote) {
+        $boltDiscountAmount = $this->getBoltDiscountAmount($transaction);
+        $quoteDiscountAmount = $quote->getShippingAddress()->getDiscountAmount();
+
+        return $this->isOnePennyRoundingError($boltDiscountAmount, $quoteDiscountAmount);
+    }
+
+    protected function getBoltDiscountAmount($transaction) {
+        $boltDiscountAmount = 0;
+
+        if(isset($transaction->order->cart->discounts)) {
+            foreach($transaction->order->cart->discounts as $discount) {
+                if(isset($discount->amount->amount) && is_numeric($discount->amount->amount) && $discount->amount->amount > 0) {
+                    $boltDiscountAmount -= $discount->amount->amount / 100;
+                }
+            }
+        }
+
+        return $boltDiscountAmount;
+    }
+
+    protected function isOnePennyRoundingError($boltDiscountAmount, $quoteDiscountAmount) {
+        if($boltDiscountAmount != $quoteDiscountAmount) {
+            $difference = round(abs($boltDiscountAmount - $quoteDiscountAmount), 2);
+            if($difference <= 0.01) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    protected function fixQuoteDiscountAmount($transaction, $quote) {
```

Comment:
Can you add comment (and maybe link to site describing magento's bug) to explain why this is needed and how it fixes the issue?

Reply:
Pushed new change to address this.